fontir/src/glyph.rs

```diff
@@ -0,0 +1,681 @@
+//! IR glyph processing.
+//!
+//! Notably includes splitting glyphs with contours and components into one new glyph with
+//! the contours and one updated glyph with no contours that references the new gyph as a component.
+
+use std::collections::{HashSet, VecDeque};
+
+use fontdrasil::{orchestration::Work, types::GlyphName};
+use indexmap::IndexSet;
+use kurbo::Affine;
+use log::{debug, trace};
+use ordered_float::OrderedFloat;
+use write_fonts::pens::{write_to_pen, BezPathPen, ReverseContourPen};
+
+use crate::{
+    coords::NormalizedLocation,
+    error::WorkError,
+    ir::{Component, Glyph},
+    orchestration::{Context, IrWork},
+};
+
+pub fn create_finalize_static_metadata_work() -> Box<IrWork> {
+    Box::new(FinalizeStaticMetadataWork {})
+}
+
+struct FinalizeStaticMetadataWork {}
+
+/// Glyph should split if it has components *and* contours.
+///
+/// Such a glyph turns into a simple glyf with the contours and a
+/// composite glyph that references the simple glyph as a component.
+///
+/// <https://learn.microsoft.com/en-us/typography/opentype/spec/glyf>
+fn has_components_and_contours(glyph: &Glyph) -> bool {
+    glyph
+        .sources
+        .values()
+        .any(|inst| !inst.components.is_empty() && !inst.contours.is_empty())
+}
+
+/// Does the Glyph uses components whose transform is different at different locations designspace?
+fn has_consistent_component_transforms(glyph: &Glyph) -> bool {
+    distinct_component_seqs(glyph).len() <= 1
+}
+
+fn name_for_derivative(base_name: &GlyphName, names_in_use: &IndexSet<GlyphName>) -> GlyphName {
+    let mut i = 0;
+    let base_name = base_name.as_str();
+    loop {
+        let new_name: GlyphName = format!("{base_name}.{i}").into();
+        if !names_in_use.contains(&new_name) {
+            return new_name;
+        }
+        i += 1;
+    }
+}
+
+fn split_glyph(glyph_order: &IndexSet<GlyphName>, original: &Glyph) -> (Glyph, Glyph) {
```

Comment:
This return type could use a struct specifying which glyph is simple and which is composite. Or at least a doc comment on the function noting such.

Reply:
Added a doc comment.